src/gui/categoryfilterproxymodel.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2017  Frédéric Brière <fbriere@fbriere.net>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "categoryfilterproxymodel.h"
+
+#include "base/utils/string.h"
+#include "categoryfiltermodel.h"
+
+CategoryFilterProxyModel::CategoryFilterProxyModel(QObject *parent)
+    : QSortFilterProxyModel(parent)
+{
+}
+
+QModelIndex CategoryFilterProxyModel::index(const QString &categoryName) const
+{
+    return mapFromSource(static_cast<CategoryFilterModel *>(sourceModel())->index(categoryName));
+}
+
+QString CategoryFilterProxyModel::categoryName(const QModelIndex &index) const
+{
+    return static_cast<CategoryFilterModel *>(sourceModel())->categoryName(mapToSource(index));
+}
+
+bool CategoryFilterProxyModel::lessThan(const QModelIndex &left, const QModelIndex &right) const
+{
+    // "All" and "Uncategorized" must be left in place
+    if (CategoryFilterModel::isSpecialItem(left) || CategoryFilterModel::isSpecialItem(right)) {
+        return left.row() < right.row();
+    }
+    else {
+        const QString &leftString = sourceModel()->data(left).toString();
+        const QString &rightString = sourceModel()->data(right).toString();
```

Comment:
@Chocobo1: Thanks for the tip.  I had blindly copied the Qt documentation; not sure why they chose this roundabout way in the first place.